epoch.go

```diff
@@ -600,9 +597,9 @@ func (e *Epoch) handleFinalizationMessage(message *Finalization, from NodeID) er
 }
 
 func (e *Epoch) handleFinalizationForPendingOrFutureRound(message *Finalization, round uint64, nextSeqToCommit uint64) {
-	if round == e.round {
+	if round <= e.round {
```

Comment:
I don't have a real objection against this change, but can you tell what made you do this? Was there a flake you encountered or a test that fails? I ran the tests and they pass when reverting the change. 

Asking because the only way we're still verifying a proposal for a past round is if we've advanced a round because we received an empty notarization, but since we received a finalization, this is a contradiction.

Reply:
> but we can still have a round that has an empty notarization & finalization?

We may have it only due to the fact that we finalize each block recursively.

But if a round was timed out by `f+1` correct nodes and an empty notarization was produced, this round is blocked from being finalized. 

We may later notarized and finalize a more advanced round and then we will recursively collect a finalization. 
However if that happens, it means we're behind, and we will replicate because we should receive the finalization with the higher round, no?